src/js/stores/preferences.js

```diff
@@ -141,17 +141,24 @@ define(function (require, exports, module) {
             _storage.setItem(PREF_INDEX, keysJSON);
             this.emit("change");
         },
-
+        
         /**
          * Set a single preference.
          *
          * @private
          * @param {{key: string, value: *}} payload
          */
-        _setPreference: function (payload) {
-            var key = payload.key,
-                value = payload.value;
+        _handleSetPreference: function (payload) {
+            this.setPreference(payload.key, payload.value);
+        },
 
+        /**
+         * Set a single preference.
+         *
+         * @param {sring} key
+         * @param {*} value
+         */
+        setPreference: function (key, value) {
```

Comment:
Reading your code a bit more, I've noticed that you're calling this function from the library store. Those store handlers are reacting to some dispatch, so what you should do instead is install handlers for those events in this store as well, and edit this store's own data structures only through those handlers.

Or you can enqueue/transfer to setPreference action from the actions that are dispatching things (if you need library store to react first, use dispatchAsync, then grab whatever data you need from the library store and enqueue a set preference action.